Missing encounters fixes

SimplyBLGDev opened the pull request:
Review commits for more details on changes.

Overall changes:

+ Encounter methods for special Pokémon (Only one and Pokéflute).
+ Encounter conditions for special Pokémon.
+ Missing area for special Pokémon (Bell Tower Roof for Ho-Oh, Sky Pillar Roof for Rayquaza, Terra Cave for Groudon, Marine Cave for Kyogre, Cave of Origin b4f for Groudon/Kyogre)
+ Locations and areas for roaming Pokémon (Kanto, Johto, Hoenn and Sinnoh)
+ Missing encounters in:

- Red, Blue, Yellow (Articuno, Zapdos, Moltres, Mewtwo, Electrode, Voltorb and Snorlax).
- Gold, Silver, Crystal (Raikou, Entei, Suicune, Lugia, Ho-Oh, Snorlax)
- Ruby, Sapphire, Emerald (Latios, Latias, Groudon, Kyogre, Rayquaza)
- Fire Red, Leaf Green (Articuno, Zapdos, Moltres, Mewtwo, Snorlax, Electrode, Raikou, Entei, Suicune)
- Heart Gold, Soul Silver (Raikou, Entei, Suicune, Lugia, Ho-Oh, Latios, Latias, Articuno, Zapdos, Moltres, Mewtwo, Snorlax)

- Encounter methods for headbutts (low encounter tree, normal encounter tree, high encounter tree)
- Headbutt encounter slots for Gold Silver Crystal
- Missing location area for headbutt encounters (Azalea town Area)
- Headbutt encounters for Gold Silver Crystal

SimplyBLGDev commented on data/v2/csv/encounters.csv:
```diff
@@ -54363,16 +54363,16 @@ id,version_id,location_area_id,encounter_slot_id,pokemon_id,min_level,max_level
+58079,3,330,562,101,43,43
+58080,3,332,562,146,50,50
+58081,3,276,580,143,30,30
+58082,3,309,580,143,30,30
```
Changed encounter slot id from 561 to 562 and 579 to 580 for crystal encounters since slots 561 and 579 are associated to version group gold-silver instead of version group crystal, this error didn't seem to generate any error anywhere but it was to be corrected anyways

Naramsim commented on data/v2/csv/encounter_condition_value_prose.csv:
```diff
@@ -39,3 +39,18 @@ encounter_condition_value_id,local_language_id,name
 19,9,During Autumn
 20,6,Im Winter
 20,9,During Winter
+21,9,Bulbasaur as starter
```
Hi @SimplyBLGDev, great work!

I was wondering why do we need to list all these "x as starter". Can't we just summarize those with a single entry "starter"?

SimplyBLGDev replied:
How would that work?

SimplyBLGDev replied:
The 'x as a starter' conditions are values for the parent condition 'starter', they need to be different condition values since different encounters require each of them individually.

Naramsim replied:
Ok, I was thinking about leaving only a general description and summarizing all those `x as a starter` with a single encounter method that would work for all of them. But your solution is actually nice since it's more specific.

I'll try to test it locally.

SimplyBLGDev replied:
Well, the encounter method doesn't change, the roaming legendaries are found by walking, so the encounter method is that, the x as starter are prerequisite for the encounter to exist, hence the encounter condition and it's values.

Naramsim replied:
Now I got it, depending on the starter, entei or suicune or raikou will appear.

SimplyBLGDev commented:
- Added *all* headbutt encounters from Gold, Silver and Crystal, taken straight from a decompilation (check the 'add heabutt encounters' commit for more info).
- Added Snorlax encounters from Gold, Silver, Crystal, Heart Gold and Soul Silver.
- Added Groudon, Kyogre and Rayquaza encounters for Ruby, Sapphire and Emerald.
- Also added some additional conditions, locations, location areas and condition values to support these new encounters.
- Corrected some minor typos that were present in earlier commits.

Please merge as soon as possible.

Naramsim commented on data/v2/csv/location_areas.csv:
```diff
@@ -682,8 +682,14 @@ id,location_id,game_index,identifier
+795,800,0,area
+796,801,0,area
+797,802,0,area
+798,228,0,area
```
You left out `799` on purpose?

SimplyBLGDev replied:
Yes, it was removed in one of the later commits since it was unused (The data was to be used on headbutts but was an empty table, hence I created the area and later realized it was useless)

Naramsim commented on data/v2/csv/encounters.csv:
```diff
@@ -54428,3 +54428,1700 @@ id,version_id,location_area_id,encounter_slot_id,pokemon_id,min_level,max_level
 58138,9,796,566,380,40,40
 58139,7,796,565,381,40,40
 58140,9,796,566,381,40,40
+58141,6,287,593,163,10,10
```
Just out of curiosity, how did you scrape these values?

SimplyBLGDev replied:
I explained in the headbutts commit, I got them from a decompilation of gold, silver and crystal and made a tool in python to process them into these tables.

Naramsim commented:
I checked a bit locally. Seems to work well! Let's check the API diff.

pokeapi-machine-user commented:
A [PokeAPI/api-data](https://github.com/PokeAPI/api-data) refresh has started. In 45 minutes the staging branch of [PokeAPI/api-data](https://github.com/PokeAPI/api-data/tree/staging) will be pushed with the new generated data. <br><br> The staging branch will be deployed in our [staging environment](https://staging.pokeapi.co/api/v2/) and you will be able to review the entire API. <br><br> A Pull Request ([master](https://github.com/PokeAPI/api-data/tree/master)<-[staging](https://github.com/PokeAPI/api-data/tree/staging)) will be also created at [PokeAPI/api-data](https://github.com/PokeAPI/api-data/pulls) and assigned to the PokeAPI Core team to be reviewed. If approved and merged new data will soon be available worldwide at [pokeapi.co](https://pokeapi.co/api/v2/).

pokeapi-machine-user commented:
The updater script has finished its job and has now opened a Pull Request towards [PokeAPI/api-data](https://github.com/PokeAPI/api-data/pulls) with the updated data. <br><br> You can see the Pull Request deployed at our [staging environment](https://staging.pokeapi.co/api/v2/) when [CircleCI deploy](https://app.circleci.com/pipelines/github/PokeAPI/deploy) will be finished (_check the started time of the last build_).
